apps/vs-code-react/src/webviewCommunication.tsx

```diff
@@ -1,46 +1,153 @@
+  changeMapDefinition,
+  changeRuntimePort,
+  changeSchemaList,
+  changeSourceSchemaFilename,
+  changeTargetSchemaFilename,
+  changeUseExpandedFunctionCards,
+  changeXsltContent,
+  changeXsltFilename,
+} from './state/DataMapSlice';
+import { initializeDesigner, updateCallbackUrl, updateFileSystemConnection, updatePanelMetadata } from './state/DesignerSlice';
+import type { InitializePayload } from './state/WorkflowSlice';
+import {
+  initialize as initializeWorkflow,
+  updateAccessToken,
+  updateTargetDirectory,
+  addStatus,
+  setFinalStatus,
+} from './state/WorkflowSlice';
+import { initialize } from './state/projectSlice';
+import type { AppDispatch, RootState } from './state/store';
+import { SchemaType } from '@microsoft/logic-apps-data-mapper';
 import { ExtensionCommand } from '@microsoft/vscode-extension';
 import useEventListener from '@use-it/event-listener';
 import type { ReactNode } from 'react';
 import React, { useEffect } from 'react';
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 import type { WebviewApi } from 'vscode-webview';
 
 const vscode: WebviewApi<unknown> = acquireVsCodeApi();
 export const VSCodeContext = React.createContext(vscode);
 
-type MessageType = InjectValuesMessage | UpdateAccessTokenMessage | UpdateExportPathMessage | AddStatusMessage | SetFinalStatusMessage;
+type DesignerMessageType = ReceiveCallbackMessage | CompleteFileSystemConnectionMessage | UpdatePanelMetadataMessage;
+type DataMapperMessageType =
+  | FetchSchemaMessage
+  | LoadDataMapMessage
+  | ShowAvailableSchemasMessage
+  | GetAvailableCustomXsltPathsMessage
+  | SetXsltDataMessage
+  | SetRuntimePortMessage
+  | GetConfigurationSettingMessage;
+type WorkflowMessageType = UpdateAccessTokenMessage | UpdateExportPathMessage | AddStatusMessage | SetFinalStatusMessage;
+type MessageType = InjectValuesMessage | DesignerMessageType | DataMapperMessageType | WorkflowMessageType;
 
 export const WebViewCommunication: React.FC<{ children: ReactNode }> = ({ children }) => {
   const dispatch: AppDispatch = useDispatch();
+  const projectState = useSelector((state: RootState) => state.project);
+
   useEventListener('message', (event: MessageEvent<MessageType>) => {
     const message = event.data; // The JSON data our extension sent
-    switch (message.command) {
-      case ExtensionCommand.initialize_frame:
-        dispatch(initialize(message.data));
-        break;
-      case ExtensionCommand.update_access_token:
-        dispatch(updateAccessToken(message.data.accessToken));
-        break;
-      case ExtensionCommand.update_export_path:
-        dispatch(updateTargetDirectory(message.data));
-        break;
-      case ExtensionCommand.add_status:
-        dispatch(addStatus(message.data));
+    if (message.command === ExtensionCommand.initialize_frame) {
+      dispatch(initialize(message.data.project));
+    }
+
+    switch (projectState?.project) {
```

Comment:
Something to consider when testing: will dispatch(initialize(message.data.project)) call update this value right away for this switch case to work?

Reply:
Yes, I assume it will do. Lets test it to make sure it does